opentelemetry/src/common.rs

```diff
@@ -161,6 +189,8 @@ pub enum Array {
     F64(Vec<f64>),
     /// Array of strings
     String(Vec<StringValue>),
+    /// Array of bytes
+    Bytes(Vec<BytesValue>),
```

Comment:
https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/common#attribute
ByteArrary is not supported in the Otel Spec for common attributes...

But it is supported for Logs data - https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#type-any and this repo also supports it already.

@lalitb @TommyCpp  also to check if I am missing something obvious.

Reply:
Good point, I've removed Array(Bytes), I don't think it was strictly necessary for this PR

Reply:
Good point @cijothomas. The common attributes (used for `resources`, `traces`, and `metrics`) are not supposed to support bytes altogether. The use-case is only for logs, but then tokio-tracing and log crate doesn't support bytes as attributes. Just wondering if the example in the description is valid in that case or not - otel-python shouldn't allow populating the resource attributes with binary values.

Reply:
I don't see why bytes values wouldn't be allowed in resources specifically. In any case, the original ask was for span attributes but see https://github.com/open-telemetry/opentelemetry-rust/pull/2471#discussion_r1902025537

Reply:
@adriangb - I think we can remove the json example with base64 values in Resource. And also remove adding the Value::Bytes variant. The SDK should allow deserialize the Logs attributes containing the binary data, probably we can have validation for that.

Reply:
> I don't see why bytes values wouldn't be allowed in resources specifically. In any case, the original ask was for span attributes but see [#2471 (comment)](https://github.com/open-telemetry/opentelemetry-rust/pull/2471#discussion_r1902025537)

Yes i didn’t realise both span and resource attributes are driven from common attributes, and so can’t have byte value.

Reply:
Probably I was not very clear why bytes values are not allowed in resources and span attributes. As this is driven by the specs - https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/README.md#attribute

```
The attribute value is either:
- A primitive type: string, boolean, double precision floating point (IEEE 754-1985) or signed 64 bit integer.
- An array of primitive type values. The array MUST be homogeneous, i.e., it MUST NOT contain values of different types.
```

and for log attributes - https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#type-any
```
```

Reply:
> I think that bit of the spec just needs updating: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/attribute-type-mapping.md#byte-sequences

At proto level, there is common structure `AnyValue` used for all kind of attributes (including logs attributes), so it supports Binary type. However at SDK level, it shouldn't be possible to populate the binary values from resouce and span-attributes into this AnyValue. Refer - https://github.com/open-telemetry/opentelemetry-proto/issues/501

Reply:
I'll open an issue to clarify the spec - I believe the movement is towards allowing bytes in trace attributes.

In any case, I made the change you've suggested 😄

Reply:
> I'll open an issue to clarify the spec - I believe the movement is towards allowing bytes in trace attributes.

I agree, it would be better to get clarity from specs.

> In any case, I made the change you've suggested.

Thanks. It would be useful for deserialization of the log attributes :)